deque/dequeList.hpp

```diff
@@ -301,44 +259,6 @@ class DequeList {
     count++;
   }
 
-  /*  void append_right(T element){
-       if (is_empty()){
-           map[0]= new T[block_size];
-           back_index = 0;
-           end_block = 0;
-           front_index = back_index;
-           map[back_index][back_index] = element;
-       }
-       else{
-           // check if the block is full
-           if (back_index == block_size -1){
-               // check if map is full on right
-               if (count/block_size >= map_size){
-                   T** new_map = new T*[map_size +1];
-                   // copying elements to new map and adding new block
-                   for (int i = 0; i < map_size; i++){
-                       new_map[i] = map[i];
-                   }
-                   new_map[map_size] = new T[block_size];
-                   delete [] map;
-                   map = new_map;
-                   map_size++;
-               }
-               // now back is first element of new block and end block is the
-   next block on the right end_block = (end_block +1) % map_size; back_index =
-   0;
-
-
-           }
-           else {
-               back_index++;
-           }
-           // Find the current end block where we need to add the element
-           map[end_block][back_index] = element;
-       }
-       count++;
-   } */
-
   T pop_left() {
```

Comment:
Incorrect Indexing in pop_left and pop_right:
 The methods pop_left and pop_right incorrectly update start_block and end_block   using modulo operations that assume the blocks wrap around within the map array. This is not a correct way to manage these indices, as it doesn't account for the actual logical sequence of blocks.  Instead, decrementing and incrementing start_block and end_block directly, while managing when to reset or wrap the indices, would be more appropriate.

Reply:
Also at function is not checking boundary conditions.